src/System.Data.SQLite/SQLiteDataReader.cs

```diff
@@ -50,9 +54,43 @@ private Task<bool> NextResultAsyncCore(CancellationToken cancellationToken)
 					int index;
 					if (parameterName is not null)
 					{
+#if NET5_0
+						static unsafe int GetParameterIndex(SqliteStatementHandle currentStatement, string parameterName)
+						{
+							var utf8Length = Encoding.UTF8.GetByteCount(parameterName) + 2;
+
+							const int c_stackAllocationThreshold = 1024;
+							byte[] pooled = null;
+							Span<byte> span = utf8Length <= c_stackAllocationThreshold ?
+								stackalloc byte[utf8Length] :
+								(pooled = ArrayPool<byte>.Shared.Rent(utf8Length));
+
+							var parameterNameBytes = span;
+							if (parameterName[0] != '@')
+							{
+								parameterNameBytes[0] = (byte) '@';
+								parameterNameBytes = parameterNameBytes.Slice(1);
+							}
+							var byteCount = Encoding.UTF8.GetBytes(parameterName.AsSpan(), parameterNameBytes);
+							parameterNameBytes[byteCount] = 0;
+
+							int index;
+							fixed (byte* spanBytes = span)
+							{
+								index = NativeMethods.sqlite3_bind_parameter_index(currentStatement, spanBytes);
+							}
+
+							if (pooled is not null)
+								ArrayPool<byte>.Shared.Return(pooled);
+
+							return index;
+						}
+						index = GetParameterIndex(m_currentStatement, parameterName);
```

Comment:
Why not inline the local function? I assumed it must have multiple `return` statements but it does not.

Reply:
Can't use `Span>` in an `async` method.